modules/dasBidAdapter.js

```diff
@@ -0,0 +1,250 @@
+  const dr = customParams.dr || deepAccess(bidderRequest, 'refererInfo.ref');
+
+  if (du) customParams.du = du;
+  if (dr) customParams.dr = dr;
+
+  const dsaRequired = deepAccess(bidderRequest, 'ortb2.regs.ext.dsa.required');
+  if (dsaRequired !== undefined) {
+    customParams.dsainfo = dsaRequired;
+  }
+
+  return {
+    customParams,
+    keyValues,
+  };
+}
+
+function buildUserIds(customParams) {
+  const userIds = {};
+  if (customParams.lu) {
+    userIds.lu = customParams.lu;
+  }
+  if (customParams.aid) {
+    userIds.aid = customParams.aid;
+  }
+  return userIds;
+}
+
+function getNpaFromPubConsent(pubConsent) {
+  const params = new URLSearchParams(pubConsent);
+  return params.get('npa') == '1';
+}
+
+function buildOpenRTBRequest(bidRequests, bidderRequest) {
+  const { customParams, keyValues } = parseParams(
+    bidRequests[0].params,
+    bidderRequest,
+  );
+  const imp = bidRequests.map((bid, index) => {
+    const sizes = getAdUnitSizes(bid);
+    const imp = {
+      id: bid.bidId,
+      tagid: bid.params.slot,
+      secure: 1,
+      ext: {
+        pos: bid.params.slotSequence || 0,
+      },
+    };
+
+    if (bid.mediaTypes?.banner) {
+      imp.banner = {
+        format: sizes.map((size) => ({
+          w: size[0],
+          h: size[1],
+        })),
+      };
+    }
+
+    return imp;
+  });
+
+  const request = {
+    id: bidderRequest.bidderRequestId,
+    imp,
+    site: {
+      id: customParams.site,
+      page: customParams.du,
+      ref: customParams.dr,
+      ext: {
+        area: customParams.area,
+        kwrd: customParams.kwrd,
+        dv: customParams.DV
+      },
+      ...bidderRequest.ortb2.site,
+    },
+    device: bidderRequest.device,
+    user: {
+      ext: {
+        ids: buildUserIds(customParams),
+      },
+    },
+    ext: {
+      network: customParams.network,
+      keyvalues: keyValues,
+    },
+    at: 1,
+    tmax: bidderRequest.timeout
+  };
+
+  if (bidderRequest.gdprConsent) {
+    request.user = {
+      ext: {
+        dsa: customParams.dsainfo,
+        npa: getNpaFromPubConsent(customParams.pubconsent),
+        localcapping: customParams.local_capping,
+        ...request.user.ext,
+      },
+    };
+    request.regs = {
+      gdpr: bidderRequest.gdprConsent.gdprApplies,
+    }
+  }
+
+  return request;
+}
+
+function interpretResponse(serverResponse) {
+  const bidResponses = [];
+  const response = serverResponse.body;
+
+  if (!response || !response.seatbid || !response.seatbid.length) {
+    return bidResponses;
+  }
+
+  response.seatbid.forEach((seatbid) => {
+    seatbid.bid.forEach((bid) => {
+      const bidResponse = {
+        requestId: bid.impid,
+        cpm: bid.price,
+        currency: response.cur || 'USD',
+        width: bid.w,
+        height: bid.h,
+        creativeId: bid.crid || bid.id,
+        netRevenue: true,
+        dealId: bid.dealid || null,
+        actgMatch: bid.ext?.actgMatch,
```

Comment:
Nie wiem, czy to ma jakieś znaczenie, ale w obecnym adapterze jest tak:
```suggestion
        actgMatch: bid.ext?.actgMatch || 0,
```